osu.Game/Online/API/APIRequest.cs

```diff
@@ -102,6 +102,13 @@ public void Fail(Exception e)
             WebRequest?.Abort();
 
             Logger.Log($@"Failing request {this} ({e})", LoggingTarget.Network);
+
+            if (!(e is OperationCanceledException) && WebRequest?.ResponseStream?.Length < 1024)
+            {
+                // in the case we fail a request, spitting out the response in the log is quite helpful.
+                Logger.Log($"Failed response was \"{WebRequest.ResponseString}\"", LoggingTarget.Network);
```

Comment:
Is there a reason you didn't make this as part of `JsonWebRequest`?

Reply:
Because it's potentially useful for all web requests.

Reply:
Then as part of `WebRequest`? My point is that this would be useful to have for projects other than osu! too.

Reply:
Maybe?